soql-standalone-parser/src/main/jflex/SoQLLexer.flex

```diff
@@ -126,6 +131,9 @@ QuotedSystemIdentifier = ":" "@"? ("-" | [:jletterdigit:])+
   {SystemIdentifier} { return token(new SystemIdentifier(yytext(), false)); }
   "`" / ":" { stringStart = pos(); yybegin(QUOTEDSYSTEMIDENTIFIER); }
 
+  {TableIdentifier} { return token(new TableIdentifier(yytext())); }
+  "" { stringStart = pos(); yybegin(QUOTEDIDENTIFIER); }
```

Comment:
This second line doesn't need to be here; it says that all input sequences are potentially starts of the QUOTEDIDENTIFIER state, which is definitely wrong.